File: Data Structures & Algorithms/sort-colors/submission-4.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int j = 0;
        int k = 0;

        for (int i = 0; i < nums.size(); i++) {
            int num = nums[i];
            nums[i] = 2;
            if (num < 2) {
                nums[j++] = 1;
            }
            if (num < 1) {
                nums[k++] = 0;
            }
        }
    }
};

File: Data Structures & Algorithms/sliding-window-maximum/submission-4.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        window = collections.deque()
        output = []
        l = -1
        for i in range(len(nums)):
            while window and nums[i] >= window[-1][0]:
                window.pop()
            window.append((nums[i], i))
            
            if i >= k-1:
                l += 1
                output.append(window[0][0])
                if window[0][1] == l: window.popleft()
        return output